docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,215 @@
+```
+
+Since the `student` is the currently active plugin, let's go ahead and prepare our model, optimizer, and scheduler like normal:
+
+```python
+student_model, optimizer, scheduler = ...
+student_model, optimizer, scheduler, train_dataloader = accelerator.prepare(student_model, optimizer, scheduler, train_dataloader)
+```
+
+From here, we need to deal with the teacher model.
+
+### The teacher model
+
+First we need to tell the `Accelerator` that the zero3/teacher ZeRO configuration should be used:
+
+```python
+accelerator.state.enable_deepspeed_plugin("teacher")
+```
+
+Doing so will disable the `"student"` plugin and enable the `"teacher"` one instead. This will update the
+DeepSpeed stateful config inside of `transformers`, and change which plugin configuration gets called when using
+`deepspeed.initialize()`, allowing us to use the no-code `Zero3Init` `deepspeed` context manager `transformers` provides:
+
+```python
+teacher_model = AutoModel.from_pretrained(...)
+teacher_model = accelerator.prepare(teacher_model)
+```
+
+Otherwise you should manually initialize them under `deepspeed.zero.Init`:
+```python
+with deepspeed.zero.Init(accelerator.deepspeed_plugin.config):
+    model = MyModel(...)
+```
+
+### Training
+
+From here, your training loop can be whatever you like, making sure that `teacher_model` is never being trained on:
+
+```python
+teacher_model.eval()
+student_model.train()
+for batch in train_dataloader:
+    with torch.no_grad():
+        output_teacher = teacher_model(**batch)
+    output_student = student_model(**batch)
+    # Combine the losses or modify it in some way
+    loss = output_teacher.loss + output_student.loss
+    accelerator.backward(loss)
+    optimizer.step()
+    scheduler.step()
+    optimizer.zero_grad()
+```
+
+## Multiple models all being trained
+
+Multiple models being trained is a more complicated scenario and is actively being looked upon by the team.
+In its current state, we assume that each model **completely disjoint** from the other during training.
+
+With this situation, we still require two [`utils.DeepSpeedPlugin`]'s to be made, however we also further
+require a second `Accelerator`, since we need to call the proper `deepspeed` engine, and `Accelerator` only
+carries one instance of it at a time.
+
+Since the [`state.AcceleratorState`] is a stateful object though, it is already aware of both `DeepSpeedPlugin`'s available,
+meaning you can just instantiate a second `Accelerator` with no extra arguments:
+
+```python
+accelerator_0 = Accelerator(deepspeed_plugin=deepspeed_plugins)
+accelerator_1 = Accelerator()
+```
+
+Similar to before, we can call either `accelerator_0.state.enable_deepspeed_plugin()` to enable/disable
+a particular plugin, and call `.prepare` like normal:
+
+```python
+# can be `accelerator_0`, `accelerator_1`, or by calling `AcceleratorState().enable_deepspeed_plugin(...)`
+accelerator_0.state.enable_deepspeed_plugin("model1")
```

Comment:
if you are using the `_\d` pattern can we use it consistently? i.e. `model_1`, etc? or w/o `_` but then again consistently w/o it.

Reply:
i think the accounting is weird too, you have model0 and model1 and model2 - something is off here - you should have only 2 models here, no?

Reply:
Fixed :)